pkg/server/server.go

```diff
@@ -773,6 +854,9 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 			return nil, serverErrors.ValidationError(err)
 		}
 
+		// Note for ListObjects:
```

Comment:
We have 3 deadlines involved here:

1) The HTTP reverse-proxy timeout deadline (`--http-upstream-timeout`)
2) the RPC handler timeout deadline (`--request-deadline`)
3) the ListObjects deadline (`--listObjects-deadline`)

The ListObjects deadline (3) should be less than the request deadline (2). If (3) is less than (2) then we can distinguish this error, and that's what should be done here. The deadline limit for searching for ListObjects results should return before the request timeout.

Deadlines (1) and (2) are related to one another, but (2) is the super of the two.

Reply:
Decided that this wouldn't be applicable now as ListObjects will return partial results when when context deadline exceeds